### PR TITLE
feat: Post contribution and question directly to the backend

### PR DESCRIPTION
**Pull request by lutangar**
This is the first draft allowing to post `Contribution` and other `Question` to the backend.
No more code sending email out there in the wild :tiger: nor more *sendinblue*.

Form submission action is sent to the background and then to the backend API to avoid CORS issues.

This actually removes a lot of code :sweat_smile: 

> EDIT: The hidden goal here is to gain back automated publication process and to ship more :shipit: 

> This PR works with https://github.com/dis-moi/backend/pull/161

### CODE REVIEW

**Comment by MaartenLMEM** [comment]
Some weeks ago, I created an issue about the same problem but can't find it. 
So here is an important point about admin needs behind this features : 
- As admin of Dismoi, I need to be alerted when there is a new contribution or information demand to a contributor. Because I want to publish it quickly, or ask contributor to answer without adding a administrative delay.
- It's would be easier if web users contributions were filled directly in BO.

Are those needs still still in the scope ? @lutangar

**Comment by lutangar** [comment]
> It's would be easier if web users contributions were filled directly in BO.

Yes this is exactly what's achieved here.

> I need to be alerted when there is a new contribution or information 

What can of alert would you expect ideally? Email? Can you think of smthing else?

**Comment by MaartenLMEM** [comment]
> What can of alert would you expect ideally? Email? Can you think of smthing else?

Ideally Email, like currently. 
A Slack alert otherwise. 
I don't see other alert means.

File: src/app/content/sagas/contribution.ts [hunk 1]
```diff
@@ -1,47 +1,44 @@
...
+      if (contributionSubmittedAction) {
+        // reset history, form and redirect to success page
+        yield put(go(-history.entries.length));
+        yield put(replace('/contribute/submitted'));
+
+        yield put(reset(form));
+
+        yield call(resolve);
+      } else {
+        // @todo we should be able to extract the error from `contributionSubmissionFailedAction`
```
Comment (JalilArfaoui) [inline src/app/content/sagas/contribution.ts:32]: Maybe `JSON.stringify` it before sending it to tab ?

Comment (lutangar) [reply src/app/content/sagas/contribution.ts:32]: Good suggestion, I'll try to modify the *"action socket"* code and see where it's going.

Comment (lutangar) [reply src/app/content/sagas/contribution.ts:32]: #816

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -222,3 +222,19 @@ extensions.lmem@gmail.com
...
 To generate your own tokens and deploy from your local environment :
 https://github.com/DrewML/chrome-webstore-upload/blob/master/How%20to%20generate%20Google%20API%20keys.md
+
+### Firefox Addons
+
+> [Developer Hub](https://addons.mozilla.org/en-US/developers/addons) 
+>
+> infrastructure@dismoi.io
+
+In order for the Mozilla review to complete successfully, please the following steps :
```
Comment (JalilArfaoui) [inline docs/CONTRIBUTING.md:232]: Why is this `extension` publishing doc added to the backend repo ?  and to this PR ?

**Review by JalilArfaoui — CHANGES_REQUESTED** [review]

**Comment by christpet** [comment]
@lutangar would love a status update here as well! Is this good to go? :)

**Review by JalilArfaoui — APPROVED** [review]
♥

**Comment by JalilArfaoui** [comment]
LGTM ! Sorry for the delay …

Branch is conflicting now 😅

**Comment by christpet** [comment]
@lutangar since this branch has conflicts, would you prefer to be the one to resolve them? Let's get this merged if possible.

**Comment by lutangar** [comment]
I'm on it.

**Review by gregoirelacoste — APPROVED** [review]
LGTM !

**Comment by christpet** [comment]
Testing on the Firefox staging extension (v3.67.2) surfaced a bug on this one:

Description of bug:
- When i try to post a contribution from the extension, it loads forever and doesn't give a success message
- The contribution also does not show up in the back-office
image


Steps to reproduce:
- Go to Firefox staging extension 
- Click "poster" and write a message with an email address not currently in the system
- Click "preview" to preview the message
- Click "submit" and wait... it does not stop loading even after 10 minutes of waiting

Does this have something to do with the Cloudflare cache @lutangar ?

**Comment by christpet** [comment]
As you can see @gregoirelacoste there is also a small bug in the i18n side - it says "Prévisualiser" at the top instead of "Preview" (re https://github.com/dis-moi/extension/pull/807 )

**Comment by christpet** [comment]
To update - here is the "ERROR" I see in the console when this bug occurs:

`[CONTENT] @ 13:31:39.171 ERROR (in 0.00 ms) redux-logger.js:1:5906
 prev state 
Object { installationDetails: {…}, ui: {…}, notices: (3) […], router: {…}, form: {…}, serviceMessage: {…}, contributors: (83) […] }
redux-logger.js:1:6179
 action     
{…}
​
"@@redux-saga/SAGA_ACTION": true
​
error: true
​
meta: Object { severity: 4 }
​
payload: DOMException: The object could not be cloned.
​
type: "ERROR"
​
<prototype>: Object { … }
redux-logger.js:1:6299
 next state 
Object { installationDetails: {…}, ui: {…}, notices: (3) […], router: {…}, form: {…}, serviceMessage: {…}, contributors: (83) […] }
redux-logger.js:1:6549`

**Comment by christpet** [comment]
Also says this:
Source map error: Error: NetworkError when attempting to fetch resource.
Resource URL: moz-extension://cf24ada4-752a-9347-933c-e67fb767e585/js/browser-polyfill.js
Source Map URL: browser-polyfill.js.map

**Comment by MaartenLMEM** [comment]
Contribution is working very well, with delicious details in the process. 

But as a question is not working. In staging firefox, I had this error message : 
image

**Comment by gregoirelacoste** [comment]
@MaartenLMEM i just created new issue from your comment https://github.com/dis-moi/extension/issues/940

**Pull request merged**